controlplane/kubeadm/controllers/scale.go

```diff
@@ -90,8 +90,8 @@ func (r *KubeadmControlPlaneReconciler) scaleDownControlPlane(
 ) (ctrl.Result, error) {
 	logger := controlPlane.Logger()
 
-	if err := r.reconcileHealth(ctx, cluster, kcp, controlPlane); err != nil {
-		return ctrl.Result{}, &capierrors.RequeueAfterError{RequeueAfter: healthCheckFailedRequeueAfter}
+	if result, err := r.reconcileHealth(ctx, cluster, kcp, controlPlane); !result.IsZero() || err != nil {
+		return result, err
 	}
```

Comment:
Alternatively, I'm wondering if it might make things easier to read and more explicit if we check for each condition independently:
```suggestion
	result, err := r.reconcileHealth(ctx, cluster, kcp, controlPlane)
	if err != nil {
		return ctrl.Result{}, err
	} else if !result.IsZero() {
		return result, nil
	}
```